specification/cognitiveservices/ContentSafetyStudio/main.tsp

```diff
@@ -0,0 +1,429 @@
+model ContentSafetyOperationQueryParameter {
+  @query
+  @doc("Content safety operation, the value could be 'AnalyzeText', 'AnalyzeImage'.")
+  contentSafetyOperation: ContentSafetyOperation;
+}
+
+@doc("Provides the 'contentSafetyVersion' query parameter for list operations.")
+model ContentSafetyVersionQueryParameter {
+  @query
+  @doc("API version of the call, the value could be '2023-04-30-preview'.")
+  contentSafetyVersion: ContentSafetyVersion;
+}
+
+@doc("Provides the 'aggregation' query parameter for list operations.")
+model AggregationQueryParameter {
+  @query
+  @doc("Aggregation method of the latency, the value could be 'Percentile99', 'Percentile95', 'Average'.")
+  aggregation: Aggregation;
+}
+
+@doc("Provides the most common query parameters for list operations.")
+model CommonListQueryParameters {
+  ...TopQueryParameter;
+  ...SkipQueryParameter;
+  ...MaxPageSizeQueryParameter;
+  ...StartTimeQueryParameter;
+  ...EndTimeQueryParameter;
+}
+
+@doc("Provides the granularity along with common query parameters for list operations.")
+model TimeGranularityListQueryParameters {
+  ...TopQueryParameter;
+  ...SkipQueryParameter;
+  ...MaxPageSizeQueryParameter;
+  ...StartTimeQueryParameter;
+  ...EndTimeQueryParameter;
+  ...TimeGranularityQueryParameter;
+}
+
+@doc("Provides the modality along with common query parameters for list operations.")
+model ModalityQueryParameters {
+  ...TopQueryParameter;
+  ...SkipQueryParameter;
+  ...MaxPageSizeQueryParameter;
+  ...StartTimeQueryParameter;
+  ...EndTimeQueryParameter;
+  ...TimeGranularityQueryParameter;
+  ...ContentSafetyVersionQueryParameter;
+  ...ModalityTypeQueryParameter;
+}
+
+@doc("Provides the content safety operation and version info along with common query parameters for list operations.")
+model ApiListQueryParameters {
+  ...TopQueryParameter;
+  ...SkipQueryParameter;
+  ...MaxPageSizeQueryParameter;
+  ...StartTimeQueryParameter;
+  ...EndTimeQueryParameter;
+  ...TimeGranularityQueryParameter;
+  ...ContentSafetyOperationQueryParameter;
+  ...ContentSafetyVersionQueryParameter;
+}
+
+@doc("Provides the aggregation query parameter along with common query parameters for list operations.")
+model AggregationListQueryParameters {
+  ...TopQueryParameter;
+  ...SkipQueryParameter;
+  ...MaxPageSizeQueryParameter;
+  ...StartTimeQueryParameter;
+  ...EndTimeQueryParameter;
+  ...TimeGranularityQueryParameter;
+  ...ContentSafetyOperationQueryParameter;
+  ...ContentSafetyVersionQueryParameter;
+  ...AggregationQueryParameter;
+}
+
+interface Metrics {
+  #suppress "@azure-tools/typespec-azure-core/use-standard-operations" "Does not fit any standard operation pattern"
+  @summary("List API request counts at different timestamps given a time range.")
+  @doc("List API request counts at different timestamps given a time range.")
+  @route("contentSafety/studio/metrics/requestCounts")
+  op listTotalCalls (
```

Comment:
Recommend to use operation in Azure.Core, e.g.
https://github.com/Azure/typespec-azure/blob/main/packages/typespec-azure-core/lib/operations.tsp#L18

Reply:
updated